src/main/java/org/checkstyle/autofix/recipe/FinalLocalVariable.java

```diff
@@ -83,33 +154,96 @@ public J.VariableDeclarations visitVariableDeclarations(
                     && !declarations.hasModifier(J.Modifier.Type.Final)) {
                 final J.VariableDeclarations.NamedVariable variable = declarations
                         .getVariables().get(0);
-                if (isAtViolationLocation(variable)) {
-                    final List<J.Modifier> modifiers = new ArrayList<>();
+                if (variable.getMarkers().findFirst(FinalLocalVariableMarker.class).isPresent()) {
+                    declarations = addFinalModifier(declarations);
+                }
+            }
+            return declarations;
+        }
+
+        @Override
+        public J.Block visitBlock(J.Block block, ExecutionContext executionContext) {
```

Comment:
I don't quite understand why we should specifically handle visiting blocks. My assumption was that any variable declaration node would be visited in `visitVariableDeclarations`?

Could you please provide an example of a variable declaration that won't be handled by `visitVariableDeclarations` but will be handled by `visitBlock`